lib/DxilPIXPasses/DxilDebugInstrumentation.cpp

```diff
@@ -396,11 +396,14 @@ static unsigned FindOrAddInputSignatureElement(
 
   if (ExistingElement == InputElements.end()) {
     auto AddedElement = llvm::make_unique<DxilSignatureElement>(sigPointKind);
+    unsigned int Index = static_cast<unsigned int>(InputElements.size());
     AddedElement->Initialize(name, hlsl::CompType::getF32(),
-                             hlsl::DXIL::InterpolationMode::Undefined, 1, 1);
+                             hlsl::DXIL::InterpolationMode::Constant, 1, 1,
+                             Index, 0);
```

Comment:
Same comment as for `SV_Position`, but depending on the SemanticKind interpretation at this SigPoint, packing location may be required (and must match connecting stage), or may need to be unassigned.

I see the existing code only used this to add VertexID and InstanceID to `VSIn` SigPoint.  `VSIn` represents a special case, handled by the input assembler, where the input row matches the element index in the signature, and elements are not packed as they would be in other stages.  You also lucked out on not breaking anything you were relying on when not supplying the `IndexVector` (which will break some things).  That's why this approach appeared to work for these values.